src/main/flight/mixer.c

```diff
@@ -743,27 +744,118 @@ STATIC_UNIT_TESTED void servoMixer(void)
 #endif
 #endif
 
+void mixConstrainMotorForFailsafeCondition(uint8_t motorIndex)
```

Comment:
Maybe making this side-effect free function (returning computed value, taking original motor command as parameter) will make the code easier to read